cookbooks/aws-parallelcluster-environment/files/default/network_interfaces/configure_nw_interface.sh

```diff
@@ -9,18 +9,28 @@
 set -e
 
 if
-  [ -z "${DEVICE_NAME}" ] ||          # name of the device
   [ -z "${DEVICE_NUMBER}" ] ||        # index of the device
   [ -z "${NETWORK_CARD_INDEX}" ] ||   # index of the network card
   [ -z "${GW_IP_ADDRESS}" ] ||        # gateway ip address
-  [ -z "${DEVICE_IP_ADDRESS}" ] ||    # ip address to assign to the interface
   [ -z "${CIDR_PREFIX_LENGTH}" ] ||   # the prefix length of the device IP cidr block
   [ -z "${NETMASK}" ]                 # netmask to apply to device ip address
 then
   echo 'One or more environment variables missing'
   exit 1
 fi
 
+# Check if this is an EFA-only interface (no device name or IP)
+if [ -z "${DEVICE_NAME}" ] && [ -z "${DEVICE_IP_ADDRESS}" ]; then
+  echo "EFA-only interface detected - skipping IP configuration"
+  exit 0
+fi
+
+# If one of these is missing but not both, it is an invalid configuration
+if [ -z "${DEVICE_NAME}" ] || [ -z "${DEVICE_IP_ADDRESS}" ]; then
+    echo "One or more environment variables missing"
```

Comment:
Please specify the name of variables that may be missing.
This script does not have -x, so if we see from the logs the message 'One or more environment variables missing' we do not know if it is the first or second check that is failing.